### PR TITLE
Replace python-jose with joserfc

### PR DESCRIPTION
**Pull request by marcospri**
For:

- https://github.com/hypothesis/lms/issues/3922

Python jose has some maintainability issues. Replacing it with joserfc which powers the authlib library.

- Replace python-jose for encryption
- Replace python-jose for signed tokens

https://github.com/authlib/joserfc Doesn't look like it hast a ton of usage based on the stars etc but it the jose backend for authlib: https://github.com/authlib/joserfc

### CODE REVIEW

File: src/h_vialib/secure/encryption.py [hunk 1]
```diff
@@ -1,26 +1,23 @@
 import json
 
-from jose import constants, jwe
+from joserfc import jwe
 
 
 class Encryption:
-    JWE_ALGORITHM = constants.ALGORITHMS.DIR
-    JWE_ENCRYPTION = constants.ALGORITHMS.A128CBC_HS256
+    JWE_ALGORITHM = "dir"
```
Comment (marcospri) [inline src/h_vialib/secure/encryption.py:7]: Same values but the lib doesn't have constant for them.

File: src/h_vialib/secure/token.py [hunk 2]
```diff
@@ -46,8 +45,9 @@ def verify(self, token):
             raise MissingToken("Missing secure token")
 
         try:
-            return jwt.decode(token, self._secret, self.TOKEN_ALGORITHM)
-        except ExpiredSignatureError as err:
```
Comment (marcospri) [inline src/h_vialib/secure/token.py:50]: We were not handling any of these independently

File: tests/unit/h_vialib/secure/token_test.py [hunk 3]
```diff
@@ -3,51 +3,76 @@
...
 from h_vialib.exceptions import InvalidToken, MissingToken
 from h_vialib.secure.token import SecureToken
 
 
 def decode_token(token_string):
-    return jwt.decode(token_string, "a_very_secret_secret", SecureToken.TOKEN_ALGORITHM)
+    return jwt.decode(token_string, "a_very_secret_secret").claims
+
+
+# This were generated by python-jose but are different than the ones generated by joserfc
```
Comment (marcospri) [inline tests/unit/h_vialib/secure/token_test.py:17]: I'll remove this after merging.

File: tests/unit/h_vialib/secure/token_test.py [hunk 4]
```diff
@@ -3,51 +3,76 @@
...
+                "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJiIjo1LCJleHAiOjE2NzE2NjcyMTB9.J1HgfeeP3e39tX1UlTmYjB4cS6OrarY3EQkuQYHOrhE",
             ),
         ],
     )
     def test_create_output(self, token, payload, expires, output):
         token_string = token.create(payload, expires=datetime.now() + expires)
 
         assert token_string == output
-
-    def test_create_works(self, token):
```
Comment (marcospri) [inline tests/unit/h_vialib/secure/token_test.py:44]: This test is a duplicate of the previous one.

**Review by acelaya — APPROVED** [review]

**Pull request merged**